src/olympia/addons/tests/test_views.py

```diff
@@ -505,6 +506,19 @@ def test_cat_sidebar(self):
                           self.addon)
 
 
+class TestICloudRedirect(TestCase):
+    @override_switch('icloud_bookmarks_redirect', active=True)
+    def test_redirect_with_waffle(self):
+        r = self.client.get('/en-US/firefox/addon/icloud-bookmarks/')
+        assert r.status_code == 302
+        assert r.get('location') == '%s/blocked/i1214/' % settings.SITE_URL
+
+    @override_switch('icloud_bookmarks_redirect', active=False)
+    def test_redirect_without_waffle(self):
+        r = self.client.get('/en-US/firefox/addon/icloud-bookmarks/')
+        assert r.status_code == 404
```

Comment:
You aren't really testing that you're calling `addon_detail` properly - i.e. it would have been nice to test with the add-on existing, that it returns the detail page correctly.